### ISSUE TITLE
flk sync (with mothership)

### ISSUE THREAD

**Issue by blaggacao** [divnix/digga#23]
**Is your feature request related to a problem? Please describe.**
Once upon a time, well into `new flake -t nrdxp/nixflk`, user wants to rejuvenate its repo state with the latest and greatest additions of the mother ship (`nrdxp/nixflk`).

Not having done a proper clone, but using the "blessed" flake template facilities, user needs to clasical-patch his source tree on unrelated histories &mdash; manually.

**Describe the solution you'd like**
`flk sync` that does `git format-patch` on the mothership since the time where `nix flake new` was invoked. It also helps applying those patches and hands off to git tooling where merge conflicts occur.

I predict direct strategic influence on the nix tooling is a precursor to enable such workflow by recording some metadata about the original template, which I believe is currently not recorded. Glory, `nix flakes` is malleable.

**Describe alternatives you've considered**
- Not everybody is familiar with the alternative forking workflow, especially since `nix flake` has the "blessed" way of using a plain templating approach.
- This kind of workflow could alternatively be supported by upstream `nix flake` directly (maybe out of scope? - at least short to mid term until this use case gradually imposes itself naturally onto the maintainers)

Once initiated a template, tracking upgrades from the mothership manually is daunting, error-prone and in-efficient. Patching support at least mitigates the latter two. Conflict resolution might remain daunting.

**Additional context**
As a side effect self documenting patches would be the only proper way to support this workflow which in turn unforgivingly raises the bar for commit messages. All the while quickly typed / badly redacted commit messages, on their own, are an infamous local maximum over the passing of times. (Big Hint to self!)

**Comment by Pacman99** [comment]
I guess this can now be solved with `sync` as an alias for `nix flake lock --update-input digga`

**Comment by blaggacao** [comment]
Agreed, the recommended way for using upstream modules or packages would be via the normal flake-based devos sharing model.

To see if there have been api changes, it might also be interesting to try to merge a template's flake.nix through git's unrelated history merge and leave such changes in staging for the user to review.

The latter might be quite useful to have a git diff supported hands on check to see (within your own file) if the API has changed.

**Comment by Pacman99** [comment]
Actually, I'm not so sure we should have an `flk sync`. I realized you can just do `flk update digga`, and I feel like it would be confusing to have an alias for that command. Updating digga isn't really anything special and a sync subcommand could make it seem different.

**Comment by blaggacao** [comment]
Let's close this, then.
